### ISSUE TITLE
Reports "All tasks should be named." whhen the task is named with no capital letter

### ISSUE THREAD

**Issue by bolinocroustibat** [ansible/ansible-lint#4351]
Environment: MacOS 15.0
Ansible version: 10.4.0

Given the following task in a playbook:

```yaml
- name: Install and update Homebrew
  hosts: local
  tasks:
    - include_role:
        name: homebrew
        tasks_from: install
    - include_role:
        name: homebrew
        tasks_from: update
```

Ansible-lint raises the following warnings:
```
WARNING  Listing 2 violation(s) that are fatal
name[missing]: All tasks should be named.
playbooks/myplaybook.yaml:6 Task/Handler: include_role name=homebrew tasks_from=install

name[missing]: All tasks should be named.
playbooks/myplaybook.yaml:9 Task/Handler: include_role name=homebrew tasks_from=update 
```

...even though, obviously, the tasks are named.

When putting a capital letter to the roles names, the fatal warning disappear.
If I understand correctly what's happening, the warning doesn't raise the right issue (capital letter), but instead says the name is missing. If I don't, this is quite confusing and should be better explained.

**Comment by Qalthos** [comment]
Huh. This is actually kind of weird.

What it's _actually_ looking for is the `name` attribute of the task itself to be present, not of the `include_role` module. So what the rule is expecting is something like

```
- name: Do the thing that the role specifies
  include_role:
    name: homebrew
    tasks_from: install
```

The fact that capitalizing the name field of `include_role` sidesteps this is unexpected, however, and should not be happening. There might be some overlap with the two name attributes that is causing trouble.
